client/extensions/woocommerce/components/price-input/index.js

```diff
@@ -21,31 +23,73 @@ import QuerySettingsGeneral from 'woocommerce/components/query-settings-general'
 
 class PriceInput extends Component {
 	static propTypes = {
-		value: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
 		currency: PropTypes.string,
 		currencySetting: PropTypes.shape( {
 			value: PropTypes.string,
 		} ),
+		initialValue: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
+		name: PropTypes.string,
+		value: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
 	};
 
-	render() {
-		const { siteId, value, currency, currencySetting } = this.props;
-		const props = {
-			...omit( this.props, [ 'value', 'currency', 'currencySetting', 'siteId', 'dispatch' ] ),
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			value: props.value,
 		};
+	}
+
+	componentWillReceiveProps( { value } ) {
+		this.setState( { value } );
+	}
+
+	resetToInitial = () => {
+		const { initialValue, name = '' } = this.props;
+		this.setState( { value: initialValue }, () => {
+			// All uses of onChange expect and event object of this shape
+			this.props.onChange( {
+				target: {
+					name,
+					value: initialValue.toString(),
+				},
+			} );
+		} );
+	};
+
+	render() {
+		const {
+			currency,
+			currencySetting,
+			initialValue,
+			siteId,
+			translate,
+			value,
+			...props
+		} = this.props;
+
 		const displayCurrency = ! currency && currencySetting ? currencySetting.value : currency;
 		const currencyObject = getCurrencyObject( value, displayCurrency );
+		let resetButton;
+		if ( initialValue ) {
+			resetButton = (
+				<Button onClick={ this.resetToInitial } compact borderless>
+					{ translate( 'Reset' ) }
+				</Button>
+			);
+		}
 		return (
-			<div>
+			<div className="price-input">
 				<QuerySettingsGeneral siteId={ siteId } />
 				{ currencyObject ? (
 					<FormCurrencyInput
 						currencySymbolPrefix={ currencyObject.symbol }
-						value={ value }
-						{ ...props }
+						currencySymbolSuffix={ resetButton }
+						value={ this.state.value }
+						{ ...omit( props, [ 'dispatch', 'moment', 'numberFormat' ] ) }
```

Comment:
Would it be appropriate to omit `translate` too?

Reply:
No need, it's already omitted by the way this.props is destructured above.